.github/workflows/ansible-test.yml

```diff
@@ -0,0 +1,40 @@
+---
+name: Ansible test
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 6 * * *'
+env:
+  NAMESPACE: scale_computing
+  COLLECTION_NAME: hypercore
+  # ANSIBLE_TEST_PREFER_PODMAN: 1
+jobs:
+  job1:
+    outputs:
+      version_matrix: ${{ steps.set-matrix.outputs.version_matrix }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - id: set-matrix
+        name: set version_matrix output
+        # run: echo "::set-output name=matrix::{\"node\":[10, 12, 14]}"
+        run: python .github/workflows/print_ansible_test_matrix.py >> $GITHUB_OUTPUT
+
+  sanity_units:
+    needs: job1
+    name: ${{ matrix.testing_type }} test ${{ matrix.ansible }}-py${{ matrix.python }}
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJSON(needs.job1.outputs.version_matrix) }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Perform sanity testing with ansible-test
+        uses: ansible-community/ansible-test-gh-action@release/v1
+        with:
+          ansible-core-version: ${{ matrix.ansible }}
+          target-python-version: ${{ matrix.python }}
```

Comment:
:bulb: idea

The docker containers used in the `ansible-test-gh-action` include all of the required python versions. We could remove this explicit option to test against all of the python versions automatically.

Unfortunately the `ansible-test-gh-action` action doesn't have a way to say "test only these python versions" or "don't test these ones". We could probably use the `pre-test-cmd` to remove the python versions we don't want to include.

Reply:
I will try to llet ansible-test auto-select ansible versions. If this works well, no need to complicate with `target-python-version: ${{ matrix.python }}`

Reply:
Hm. ansible 2.14 tries to run units test on python 2.7, and this fails. I do not want to uninstall whatever is in container just to prevent this test combination. Easier to explicitly list what we want.

Reply:
I think adding "config.yml" in tests folder with content:
modules:
  python_requires: ">=3.8"
should skip unit tests with python2.7. When I run ansible-test units I get: "WARNING: Skipping unit tests on Python 2.7 because it is not supported by this collection. Supported Python versions are: 3.8, 3.9, 3.10, 3.11"

Reply:
Are you sure that is because of the config or is it because you don't have those versions installed on your machine? The test action has all of the python versions installed.

Reply:
I was hoping the `pre-test-cmd` option would allow us to remove python versions from the docker containers, but unfortunately it runs before the containers are created. :disappointed:

Reply:
> Hm. ansible 2.14 tries to run units test on python 2.7, and this fails. I do not want to uninstall whatever is in container just to prevent this test combination. Easier to explicitly list what we want.

I tried a handful of ways to prevent testing older python versions with the `ansible-test-gh-action` action. It seems like it simply can't be done without setting the python version explicitly as you are doing.

  I think that is what ultimately deterred me from using the action in [my testing branch](https://github.com/shoriminimoe/HyperCoreAnsibleCollection/tree/matrix-testing). I found it easier to use the `setup-python` action with the versions I care about then use `ansible-test --venv` to test all of the python versions automatically.